### PR TITLE
Starting expansion of PyGRB post-processing workflow generator

### PR DESCRIPTION
**Pull request by pannarale**
Adding two utils functions within `pycbc_pygrb_pp_workflow` and (re)designing the opening page for a PyGRB results webpage.

## Motivation
This PR continues work started with https://github.com/gwastro/pycbc/pull/4872 to breakdown the workflow changes that produce a complete [PyGRB results page](https://ldas-jobs.ligo.caltech.edu/~francesco.pannarale/LVC/pygrb_sep2024_2/).  It is the first of a series of PRs that affects `pycbc_pygrb_pp_workflow`.

## Standard information about the request
This is a: the beginning of work to allow an end-to-end PyGRB run from master, with all results displayed in the final webpage.

This change affects: PyGRB

This change changes: result presentation / plotting.

## Contents
The two new utility functions are:
1. `fish_label_in_filelist` – This is needed to retrive a specific injections results `File`, given the injection set label. E.g., when a plot for "NSBH" injections is being produce, the "NSBH" injections results `File` needs to be pinpointed among all injections results files in a `FileList`.  The function returns the `File`, but we could think of returning a dictionary `{injection set label: corresponding File}` so that we `filter` the `FileList` once and for all rather than anytime results for and injections set are needed.  Or something completely different: I am happy to hear suggestions from the PR reviewer.
2. `display_seg_plot` – The PyGRB segments plot is produced in preprocessing.  I would like the results webpage to display this in the opening page, so this function retrieves it and add metadata to it (prior to sticking it on the opening page). The metadata is added here because the current preprocessing code does not use `save_fig_with_metadata` for this plot, nor a stand alone executable to produce it.  Since PyGRB will have to undergo a pretty large review, I am currently staying away from touching the old and already reviewed preprocessing material.  In the longer term, the clean solution I have in mind is to add the capability to display the PyGRB offsource and external trigger time on top of the standard all-sky segments plot; then it would be a matter of retrieving that `File` without the need for `display_seg_plot`

The rest of the changes simply organise the layout of the landing page.

## Testing performed
The complete set of changes will produce the end result linked above, but the single PRs will enable only part of it ad do not have stand alone examples/tests.

## Additional notes
The PRs that will come up after this one will "just" add tabs and sections to the results webpage and fill them up.  These are the only 2 functions that were added.

- [x] The author of this pull request confirms they will adhere to the [code of conduct](https://github.com/gwastro/pycbc/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: bin/pygrb/pycbc_pygrb_pp_workflow [hunk 1]
```diff
@@ -43,6 +43,48 @@ __date__ = pycbc.version.date
 __program__ = "pycbc_pygrb_pp_workflow"
 
 
+# Function used to retrive a File instance from a FileList based on an
+# injection set label
+def fish_label_in_filelist(label, filelist):
```
Comment (spxiwh) [inline bin/pygrb/pycbc_pygrb_pp_workflow:48]: This function doesn't belong here. It should be a method in the FileList class (in core.py). BUT first, does the functionality desired already exist? 

Are you basically asking here if `label` is a `tag` in the file? (Note that using the file `name` is dangerous and should not be done, checking the attributes is better). If so, this exists:

https://github.com/gwastro/pycbc/blob/master/pycbc/workflow/core.py#L1549

if not, can you elaborate on what label is (or what the file attributes are, to figure out how it decides this).

File: bin/pygrb/pycbc_pygrb_pp_workflow [hunk 2]
```diff
@@ -43,6 +43,48 @@ __date__ = pycbc.version.date
...
+                        in x.name.lower(), filelist))
+        out_file = out_file[0] if len(out_file) == 1 else None
+
+    return out_file
+
+
+# Function to retrieve the segments plot (produced in the preprocessing stage)
+# that ensures its copy is saved in the output directory before returning to
+# the original working directory. Appropriate meta data is added to the plot.
+def display_seg_plot(output_dir, segments_dir):
```
Comment (spxiwh) [inline bin/pygrb/pycbc_pygrb_pp_workflow:64]: Do we need to use `glob`s here? If the segment file name is predictable, can we just explicitly look for it? (Bonus points if the name is passed through explicitly, either as a file object, or (if the preprocessing stage is a separate executable) as a path on the command line on in the ini file?

File: bin/pygrb/pycbc_pygrb_pp_workflow [hunk 3]
```diff
@@ -139,25 +183,25 @@ logging.info("Created log file %s", wf_log_file.storage_path)
...
-files = _workflow.FileList([])
 
 #
-# Create information table about the GRB trigger and plot the search grid
+# Opening page
 #
-info_table_node, grb_info_table = _workflow.make_info_table(wflow, 'pygrb_grb_info_table', out_dir)
+# Create the information table about the GRB trigger
+info_table_node, grb_info_table = \
+    _workflow.make_pygrb_info_table(wflow, 'pygrb_grb_info_table', out_dir)
```
Comment (spxiwh) [inline bin/pygrb/pycbc_pygrb_pp_workflow:192]: Run this one through `black` to get correct formatting .... At some point we'll figure out how to automate this.

Comment (spxiwh) [reply bin/pygrb/pycbc_pygrb_pp_workflow:192]: (Actually, might make sense to do a pass over this code with `black` *after* you've merged in the patches you're working on, and have that as a separate patch).

**Review by spxiwh — CHANGES_REQUESTED** [review]
Main question is about the `File in FileList` function. I would like to understand what this is testing and would like it moved to a method of a FileList class (unless this is doing something very specific/odd or the functionality already exists).

File: bin/pygrb/pycbc_pygrb_pp_workflow (hunk 3 above)
Comment (pannarale) [reply bin/pygrb/pycbc_pygrb_pp_workflow:203]: I was going to suggest to keep this for the very end, yes.

**Comment by pannarale** [comment]
I followed the suggestions above, adopting the solutions discussed privately.  The webpage and all its contents were successfully regenerated [here](https://ldas-jobs.ligo.caltech.edu/~francesco.pannarale/LVC/pygrb_sep2024_3).

**Review by spxiwh — APPROVED** [review]

**Pull request merged**